src/main/java/com/suse/salt/netapi/calls/LocalCall.java

```diff
@@ -101,6 +145,76 @@ public LocalCall(String functionName, Optional<List<?>> arg,
     }
 
     /**
+     * Calls this salt call via the async client and returns the results
+     * as they come in via the event stream.
+     *
+     * @param client SaltClient instance
+     * @param target the target for the function
+     * @param events the event stream to use
+     * @param cancel future to cancel the action
+     * @return a map from minion id to future of the result.
+     * @throws SaltException if anything goes wrong
+     */
+    public Map<String, CompletionStage<Result<R>>> callAsync(
```

Comment:
we could do that but the only way to get an event stream is by token authentication so if you pass it an event stream you should be already authenticated. We could do the username password thing instead of the event stream and then creating the event stream ourselfs. What do you think?